data/mm/macros.yml

```diff
@@ -4,7 +4,7 @@
 "can_play(song)": "has(OCARINA) && has(song)"
 "can_break_boulders": "has_explosives || has(MASK_GORON)"
 "can_use_lens": "has(MAGIC_UPGRADE) && has(LENS)"
-"has_explosives": "has(BOMB_BAG) || has(MASK_BLAST)"
+"has_explosives": "has(BOMB_BAG) || (has(MASK_BLAST) && has(SHIELD))"
```

Comment:
The player is always assumed to have the shield since you can buy one in clock town (same is true for OoT), so shield checks are only for mirror shield.

However this is valuable if we add entrance rando in the future.
I suggest you add a `"has_shield": "true"` macro to mark the spots where a shield is logically required, so we can change the macro later if necessary.